database/database.go

```diff
@@ -140,41 +135,42 @@ func (db *DB) UserByID(ctx context.Context, id string) (*User, error) {
 	return &u, nil
 }
 
-// UserCreate creates a new user in the DB. We need the user object to be passed
-// by reference because we need to be able to update the ID of new user.
-func (db *DB) UserCreate(ctx context.Context, u *User) error {
-	if !u.Email.Validate() {
-		return ErrInvalidEmail
-	}
-	// Check for an existing user with this email.
-	users, err := db.managedUsersByField(ctx, "email", string(u.Email))
+// UserCreate creates a new user in the DB.
+func (db *DB) UserCreate(ctx context.Context, sub string, tier int) (*User, error) {
+	// Check for an existing user with this sub.
+	users, err := db.managedUsersByField(ctx, "sub", sub)
 	if err != nil && !errors.Contains(err, ErrUserNotFound) {
-		return errors.AddContext(err, "failed to query DB")
+		return nil, errors.AddContext(err, "failed to query DB")
 	}
 	if len(users) > 0 {
-		return ErrEmailAlreadyUsed
+		return nil, ErrUserAlreadyExists
+	}
+	u := &User{
+		ID:   primitive.ObjectID{},
```

Comment:
Have you inspected your mongo collection locally? This looks weird, every document in mongo has a field called `_id` which is its primary key. I think you might be defining users that look like this:

```
{
      _id: ObjectId(),
      ID: ObjectId(),
      Sub: sub,
      Tier: tier,
}
```

Also keep in mind that I haven't seen anywhere where we configure indices, so both `ID` and `Sub` won't have an index for them, resulting in a full COLLSCAN on every request. That's pretty detrimental to performance. 

In mongo you can use 'explain' to see the query stage pipeline, if you are using robo3t you can just do `db.collection('users').find().explain()`. You can look at the `winningplan` to see whether it's making proper use of an index or not.

Reply:
Oh I see how the ID -> _id works, you specify it using `bson`. Ignore that, the rest still applies

Reply:
I don't really mind but I think it's a bit weird we have endless discussions on database performance, with manual benchmark comparisons, and then not treat a missing index as thé highest priority.

As long as there's an issue for it and it's labeled as high prio I'm fine with anything.

Reply:
The reason we discuss DB performance and benchmark is because we're more or less locking ourselves to that DB. If not by design then by the effort of switching. So, it's a very good idea to research before making the choice.
But this performance consideration is a long-term one, for when we have tens of thousands of users. It doesn't really matter for the first two weeks.

I'll open a F/U for this (high prio): https://github.com/NebulousLabs/skynet-accounts/issues/18